plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -259,9 +259,11 @@ private static Map<String, Object> serializeStructType(ColumnMappingMode columnM
         fields.put("type", "struct");
         fields.put("fields", rowType.getFields().stream()
                 .map(field -> {
+                    String fieldName = field.getName()
+                            .orElseThrow(() -> new TrinoException(DELTA_LAKE_INVALID_SCHEMA, "Anonymous row type is not supported in Delta Lake connector"));
```

Comment:
.orElseThrow(() -> new TrinoException(DELTA_LAKE_INVALID_SCHEMA, "Anonymous field row type is not supported in Delta Lake connector"));